wikipedia/thing_description-english.md

```diff
@@ -10,7 +12,119 @@ Thing Description is a digital document that describes the metadata and interfac
+
+This example also specifies the <code>basic</code> security scheme, requiring a username and password for access. A security scheme is first given a name and its corresponding scheme in the <code>securityDefinitions</code> and then activated by specifying that name in a <code>security</code> section. In combination with the use of the HTTP protocol this example demonstrates the use of HTTP Basic Authentication.
+
+
+Below is the same connected lamp but using [[MQTT]] protocol and no security.<syntaxhighlight lang="json-ld" line="1">
+{
+    "@context": [
+        "https://www.w3.org/2019/wot/td/v1",
+        {"mqv": "http://www.example.org/mqtt-binding#"}
+    ],
+    "id": "urn:dev:ops:32473-WoTLamp-1234",
+    "title": "MyLampThing",
+    "securityDefinitions": {
+        "nosec_sc": {"scheme": "nosec"}
+    },
+    "security": ["nosec_sc"],
+    "properties": {
+        "status" : {
+            "type": "string",
+            "forms": [{
+                "href": "mqtt://mylamp.example.com/status",
+                "mqv:controlPacketValue": "SUBSCRIBE"
+            }]
+        }
+    },
+    "actions": {
+        "toggle" : {
+            "forms": [{
+                "href": "mqtt://mylamp.example.com/toggle",
+                "mqv:controlPacketValue": "PUBLISH"
+            }]
+        }
+    },
+    "events":{
+        "overheating":{
+            "data": {"type": "string"},
+            "forms": [{
+                "href": "mqtt://mylamp.example.com/oh",
+                "mqv:controlPacketValue": "SUBSCRIBE"
+            }]
+        }
+    }
+}
+</syntaxhighlight>Differently from the last TD, here the forms include MQTT protocol as specified by the WoT Binding Templates<ref name=":4">{{Cite web|url=https://www.w3.org/TR/wot-binding-templates/Overview.html|title=Web of Things (WoT) Protocol Binding Templates|website=www.w3.org|access-date=2019-11-17}}</ref>. More specifically, lines 17, 25 and 34 describe what message types should be used to use the affordances. For example, instead of HTTP GET and longpoll subprotocol to observe the overheating event, a client can subscribe to this event using the MQTT protocol. <br />
```

Comment:
we should mention, that the MQTT-based lamp service is both, a MQTT publisher and a MQTT subscriber.